src/bp/ui-studio/src/web/views/OneFlow/diagram/index.tsx

```diff
@@ -154,7 +155,12 @@ class Diagram extends Component<Props> {
     )
     this.diagramEngine.registerNodeFactory(new FailureWidgetFactory())
     this.diagramEngine.registerLinkFactory(new DeletableLinkFactory())
-    this.diagramEngine.registerNodeFactory(new PromptWidgetFactory())
+    this.diagramEngine.registerNodeFactory(
+      new PromptWidgetFactory({
+        ...commonProps,
+        getCurrentLang: () => this.getStateProperty('currentLang')
```

Comment:
Could be moved to common props (even if trigger doesn't use it currently), so we can simply pass (commonProps)